File: server/HabitTracker.Api/DTOs/CompleteHabitResponse.cs
namespace HabitTracker.Api.DTOs;

public sealed class CompleteHabitResponse
{
    public required HabitCompletionResponse Completion { get; set; }

    public IReadOnlyList<HabitAttributeRewardResponse>
        Rewards
    { get; set; }
            = Array.Empty<HabitAttributeRewardResponse>();
}

File: client/src/types/CompleteHabitResponse.ts
import type { HabitAttributeRewardResponse } from './HabitAttributeRewardResponse'
import type { HabitCompletionResponse } from './HabitCompletionResponse'

export type CompleteHabitResponse = {
  completion: HabitCompletionResponse
  rewards: HabitAttributeRewardResponse[]
}
